packages/babel/types/index.d.ts

```diff
@@ -0,0 +1,131 @@
+   * @default undefined;
+   */
+  include?: FilterPattern;
+  /**
+   * A minimatch pattern, or array of patterns, which specifies the files in the build the plugin should ignore. When relaying on Babel configuration files you can only exclude additional files with this option, you cannot override what you have configured for Babel itself.
+   * @default undefined;
+   */
+  exclude?: FilterPattern;
+  /**
+   * An array of file extensions that Babel should transpile. If you want to transpile TypeScript files with this plugin it's essential to include .ts and .tsx in this option.
+   * @default ['.js', '.jsx', '.es6', '.es', '.mjs']
+   */
+  extensions?: string[];
+  /**
+   * It is recommended to configure this option explicitly (even if with its default value) so an informed decision is taken on how those babel helpers are inserted into the code.
+   * @default 'bundled'
+   */
+  babelHelpers?: 'bundled' | 'runtime' | 'inline' | 'extends';
+  /**
+   * Before transpiling your input files this plugin also transpile a short piece of code for each input file. This is used to validate some misconfiguration errors, but for sufficiently big projects it can slow your build times so if you are confident about your configuration then you might disable those checks with this option.
+   * @default false
+   */
+  skipPreflightCheck?: boolean;
+}
+
+interface RollupBabelOutputPluginOptions
+  extends Omit<babelCore.TransformOptions, 'include' | 'exclude'> {
+  /**
+   * A minimatch pattern, or array of patterns, which specifies the files in the build the plugin should operate on. When relying on Babel configuration files you cannot include files already excluded there.
+   * @default undefined;
+   */
+  include?: FilterPattern;
+  /**
+   * A minimatch pattern, or array of patterns, which specifies the files in the build the plugin should ignore. When relaying on Babel configuration files you can only exclude additional files with this option, you cannot override what you have configured for Babel itself.
+   * @default undefined;
+   */
+  exclude?: FilterPattern;
+  /**
+   * An array of file extensions that Babel should transpile. If you want to transpile TypeScript files with this plugin it's essential to include .ts and .tsx in this option.
+   * @default ['.js', '.jsx', '.es6', '.es', '.mjs']
+   */
+  extensions?: string[];
+  /**
+   * A boolean value indicating whether to bundle in the Babel helpers.
+   * @default false
+   */
+  externalHelpers?: boolean;
+  /**
+   * An array which gives explicit control over which babelHelper functions are allowed in the bundle.
+   * @default undefined
+   */
+  externalHelpersWhitelist?: string[];
+  /**
+   * @default false
+   */
+  runtimeHelpers?: boolean;
+  /**
+   * Use with other formats than UMD/IIFE.
+   * @default false
+   */
+  allowAllFormats?: boolean;
+}
+
+type CustomInputPluginOptions = (
+  options: RollupBabelInputPluginOptions & Record<string, any>
```

Comment:
TS is not my strongest suit so take my comment with a grain of salt - maybe instead of accepting `& Record<string, any>` here this should be parametrized with a generic?

Reply:
I didn't think the `customOptions` should be a 1-1 mapping, as that would limit the custom plugin, so they could not remove or add options. I think this way the user could build stronger on top if necassary.